pkg/render/logstorage/linseed/linseed.go

```diff
@@ -310,6 +310,11 @@ func (l *linseed) linseedDeployment() *appsv1.Deployment {
 		envVars = append(envVars, corev1.EnvVar{Name: "LINSEED_EXPECTED_TENANT_ID", Value: l.cfg.Tenant.Spec.ID})
 		envVars = append(envVars, corev1.EnvVar{Name: "BACKEND", Value: "elastic-single-index"})
 		envVars = append(envVars, corev1.EnvVar{Name: "LINSEED_MULTI_CLUSTER_FORWARDING_ENDPOINT", Value: fmt.Sprintf("https://tigera-manager.%s.svc:9443", l.cfg.Tenant.Namespace)})
+		envVars = append(envVars, corev1.EnvVar{Name: "ELASTIC_DYNAMIC_INDEX_CREATION_ENABLED", Value: "true"})
+		for _, data := range l.cfg.Tenant.Spec.DataRetention {
+			envVars = append(envVars, corev1.EnvVar{Name: fmt.Sprintf("ELASTIC_%s_INDEX_NAME", strings.ToUpper(data.Type)), Value: data.IndexName})
```

Comment:
Generally it's a not-so-great pattern to link up the user-facing API (the Tenant CR) with the implementation detail (the env var). 

Even if they can be programatically generated, it's better to not do that in case we come upon a case where the strings.ToUpper isn't a good enough translation.

Tying API -> implementation detail is almost always a bad thing.

One option would be to do something like this:

```go
type DataType string

const (
  DataTypeFlowLog DataType = "FlowLogs"
)

func (t DataType) EnvVars() []corev1.EnvVar {
  return []corev1.EnvVar{
    {Name: t.IndexEnvName(), Value: data.IndexName},
  }
}

func (t DataType) IndexEnvName() string {
  switch t {
    case DataTypeFlowLog:
      return "ELASTIC_FLOW_LOG_INDEX_NAME"
    default:
      panic("Unexpected data type")
  }
}
```

WDYT?